bindings/rust/bench/benches/handshake.rs

```diff
@@ -16,16 +18,27 @@ fn bench_handshake_for_library<T: TlsConnection>(
     kx_group: KXGroup,
     sig_type: SigType,
 ) {
+    // make configs before to save time
+    let crypto_config = CryptoConfig::new(CipherSuite::default(), kx_group, sig_type);
+    let client_config_res = T::make_config(Mode::Client, crypto_config, handshake_type);
+    let server_config_res = T::make_config(Mode::Server, crypto_config, handshake_type);
+
     // generate all harnesses (TlsConnPair structs) beforehand so that benchmarks
     // only include negotiation and not config/connection initialization
     bench_group.bench_function(T::name(), |b| {
         b.iter_batched_ref(
-            || {
-                TlsConnPair::<T, T>::new(
-                    CryptoConfig::new(CipherSuite::default(), kx_group, sig_type),
-                    handshake_type,
-                    Default::default(),
-                )
+            || -> Result<TlsConnPair<T, T>, Box<dyn Error>> {
```

Comment:
not in scope for this PR but I wonder if rather than having our benchmarks be tolerant to failure (which seems to complicate the code flow a little bit) we could just have a list of denylisted commits in the historical benchmarking script? So rather than trying to benchmark a commit and failing, we should just skip it, since it's the same commits that fail every time.

Reply:
The problem is that it's the fail-safeness of this is mainly for the commits that have some config combinations succeed and some config combinations fail. The commits that just don't compile altogether already aren't included in historical benchmarking, and that's from the script being fail-safe, not the benches.